### ISSUE TITLE
[WotLK] Prince Toreth Won't Stand Still

### ISSUE THREAD

**Issue by Enverex** [cmangos/issues#912]
Even during dialogue (e.g. when issuing a quest) he will continue walking around meaning you either have to try and move with him, or accept the quest as quickly as possible otherwise he'll put too much distance between you and him, breaking dialogue.

**Comment by Grz3s** [comment]
old problem....
player interacting with any npc with randomovement. should stop him for ~2mins... before he will go back to his routine.

Problably when ranodmovement  system was redone this was damaged then ... sot sure

**Comment by killerwife** [comment]
@al3xc1985 check if this is still a thing

**Comment by al3xc1985** [comment]
Ok @killerwife , I will check it tonight and come back with proper info

**Comment by al3xc1985** [comment]
@killerwife yes, problem persist, and I can tell u also that this bug is also in the beggining quests in the dk zone with the instructior. u need to walk with him to get or give the quest... its old problem as @Grz3s said...
Npc's must have a sort of 1 or 2 min pause in their movement script for interacting with player

**Comment by al3xc1985** [comment]
and this thing I remember I noticed in many places, where u get the quest or give it back to npcs that are in movement time
